build/builder.go

```diff
@@ -505,15 +505,44 @@ func (o *Options) FindAllShards() []string {
 
 // IgnoreSizeMax determines whether the max size should be ignored.
 func (o *Options) IgnoreSizeMax(name string) bool {
+	matched := false
 	for _, pattern := range o.LargeFiles {
```

Comment:
We could bucket patterns to distinguish negate/exclude patterns from include patterns. Then we could visit each and every exclude pattern before any include pattern, just to ensure a file is not disallowed by the former. I don't think this optimization would save us much though.